src/app/view/_routes/editorganization/editorganization.component.ts

```diff
@@ -145,7 +165,61 @@ export class EditOrganizationComponent implements OnInit {
       console.log(this.organizationBuilder.build());
       this.organizationService
         .editOrganization(this.organizationBuilder.build())
-        .subscribe();
+        .subscribe((response: HttpResponse<Organization>) => {
+          console.log(response);
+        });
     }
   }
+  /*
+    Add administrator email and role to the administrators array defined above
+  */
+  addAdmin(): void {
+    let adminData: AdminType;
+    if (this.formArray?.value[2].adminRole === 'Manager') {
+      // create AdminType based on the value of mat-select field
+      adminData = {value: '2', viewValue: 'Manager'};
+    } else {
+      adminData = {value: '3', viewValue: 'Viewer'};
+    }
```

Comment:
questo if non va bene

Reply:
l'ho sostituito con uno switch

Reply:
switch è peggio di if :( l'unico posto in cui può esserci logica di questo tipo (inteso come "mapping di qualcosa su qualcos'altro") è l'abstract factory, in tutto il resto del codice bisogna usare o enum o polimorfismo perché altrimenti si viola l'open/closed principle (O di SOLID)

adesso che hai fatto l'enum, in teoria basta una di queste due:

1. castare quel valore a AdminType con `as`
2. mettere direttamente come valore nel form un'AdminType invece di una string

io tenderei verso la 2